feed/utils.js

```diff
@@ -0,0 +1,68 @@
+async function getActivityFeedData(query = {}, nextLink) {
+  validateQuery(query);
+  let finalUrl =
+    API_BASE_URL + (nextLink || '/logs' + generateActivityFeedParams(query));
+  const res = await fetch(finalUrl, {
+    credentials: 'include',
+    method: 'GET',
+    headers: {
+      'Content-type': 'application/json',
+    },
+  });
+  return await res.json();
+}
+
+function generateActivityFeedParams(query) {
+  let queryParam = 'dev=true&format=feed';
+  if (query?.category) queryParam += getLogTypesFromCategory(query?.category);
+  return `?${queryParam}`;
+}
+
+function getLogTypesFromCategory(category) {
+  switch (category) {
+    case CATEGORY.ALL:
+      return '';
+    case CATEGORY.TASK:
+      return `&type=${logType.TASK}`;
+    case CATEGORY.EXTENSION_REQUESTS:
+      return `&type=${logType.EXTENSION_REQUESTS}`;
+    case CATEGORY.TASK_REQUESTS:
+      return `&type=${logType.TASK_REQUESTS}`;
+    case CATEGORY.OOO:
+      return `&type=${logType.REQUEST_CREATED},${logType.REQUEST_REJECTED},${logType.REQUEST_APPROVED}`;
+    default: return '';
+  }
+}
+
+function addEmptyPageMessage(container) {
+  const emptyPageMessage = createElement({
+    type: 'p',
+    attributes: { class: 'page-message' },
+    innerText: 'No logs to show!',
+  });
+  container.appendChild(emptyPageMessage);
+}
+
+function addErrorElement(container, error) {
+  const errorHeading = createElement({
+    type: 'h4',
+    innerText: `Error: An error occurred while fetching logs. Please try again later.`,
+  }); 
+  const errorText = createElement({
+    type: 'p',
+    innerText: `${error}`,
+  });
+  container.appendChild(errorHeading);
+  container.appendChild(errorText);
+}
+
+
+function validateQuery(queryObject) {
```

Comment:
The function always returns true, even if it throws an error due to missing parameters. This could lead to incorrect behavior if the caller doesn't handle the error properly.

Reply:
Didn't understand this point.